lesson18/src/js/main.js

```diff
@@ -0,0 +1,275 @@
+    const button = document.createElement("button");
+    button.classList.add("arrow", `${element}`);
+    button.id = `js-button_${element}`;
+    button.textContent = `${element}`;
+    button.setAttribute("aria-label", `${element}`);
+    element === 'previous' && (button.disabled = 'true');
+    fragment.appendChild(button);
+  });
+
+  const slideArea = document.querySelector(".mainvisual__images");
+  slideArea.appendChild(buttonArea).appendChild(fragment);
+}
+
+function renderPagenation(item) {
+  const pagenation = createElementWithClassName("div", "slider__pagination");
+  const fragment = document.createDocumentFragment();
+  const current = document.createElement("span");
+  current.id = "js-current";
+  current.textContent = 1;
+  fragment.appendChild(current);
+
+  const separation = createElementWithClassName("span", "separation");
+  separation.textContent = "/";
+  fragment.appendChild(separation);
+
+  const total = document.createElement("span");
+  total.id = "js-total";
+  total.textContent = item;
+  fragment.appendChild(total);
+
+  const textArea = document.querySelector(".mainvisual__textarea");
+  textArea.appendChild(pagenation).appendChild(fragment);
+}
+
+function renderPagenationBullet(item){
+  const pagenationArea = createElementWithClassName("div","mainvisual__pagenation__bullet");
+  const fragment = document.createDocumentFragment();
+  for(let i = 0; i < item; i++){
+    const pagenationBullet = createElementWithClassName("span", "bar");
+    pagenationBullet.style.width = "calc(width / item)";
+    pagenationBullet.setAttribute ("role", "button");
+    pagenationBullet.dataset.pagenationIndex = i + 1;
+    fragment.appendChild(pagenationBullet);
+
+    i === 0 && pagenationBullet.setAttribute("aria-current","true");
+
+  }
+  const slideArea = document.querySelector('.mainvisual__images__inner');
+  slideArea.appendChild(pagenationArea).appendChild(fragment);
+}
+
+function addNoimage(){
+  const slideImageArea = document.querySelector(".mainvisual__images__inner");
+  const sliderImage = createElementWithClassName("img", "slider__image");
+  sliderImage.src = './images/slide-image01.png'
+  sliderImage.width = 800;
+  sliderImage.height = 800;
+  sliderImage.style.opacity = 1;
+  slideImageArea.appendChild(sliderImage);
+}
+
+function switchPagination(currentIndex) {
+  const pagenationCurrent = document.getElementById("js-current");
+  pagenationCurrent.textContent = currentIndex;
+
+}
+
+function switchImg(target) {
+  const active = document.querySelector(".is-active");
+  active.classList.remove("is-active");
+  const changeSlide = document.querySelector(`[data-slide-index="${target}"]`);
+  changeSlide.classList.add("is-active");
+}
+
+function switcPagenationBullet(target){
+  const pagenationBullet = document.querySelector('[aria-current="true"');
+  pagenationBullet.setAttribute('aria-current', false);
+  const pageNations = document.querySelector(`[data-pagenation-index="${target}"]`);
```

Comment:
[ref](https://getbootstrap.com/docs/4.0/components/pagination/)

pagination is name itself. not page and nation i think

```suggestion
  const paginations = document.querySelector(`[data-pagination-index="${target}"]`);
```

Reply:
Thank you for your review.
Fixed all spelling mistakes.

[d137003](https://github.com/nyattsu72/hanson-frontend-cource/pull/26/commits/d137003ce0f2c77e2c22a737b1ef8ac77d843102)